### ISSUE TITLE
run `bindinfo/bind` test will cause the tidb-server to crash

### ISSUE THREAD

**Issue by purelind** [pingcap/tidb#49189]
## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)
https://github.com/PingCAP-QE/ci/blob/main/pipelines/pingcap/tidb/latest/ghpr_check2.groovy#L80C30-L80C60
The integrationtest_with_tikv.sh could be found in here https://github.com/PingCAP-QE/ci/blob/main/scripts/pingcap/tidb/integrationtest_with_tikv.sh

### 2. What did you expect to see? (Required)
test passed

### 3. What did you see instead (Required)
test log
```shell
mysql] 2023/12/05 12:07:11 packets.go:37: unexpected EOF
time="2023-12-05T12:07:11+08:00" level=error msg="run test [bindinfo/bind] err: sql:drop global binding for delete from t1 where b = 1 and c > 1;: run \"drop global binding for delete from t1 where b = 1 and c > 1;\" at line 329 err invalid connection"
time="2023-12-05T12:08:12+08:00" level=error msg="open db failed dial tcp 127.0.0.1:46315: connect: connection refused, take time 1m0.083358173s"
time="2023-12-05T12:08:12+08:00" level=fatal msg="Open db err dial tcp 127.0.0.1:46315: connect: connection refused"
./run-tests.sh: line 1: kill: (1645) - No such process
script returned exit code 1
```
tidb-server log
```shell
[2023/12/05 12:07:11.846 +08:00] [INFO] [split_region.go:236] ["split regions complete"] ["region count"=1] ["region IDs"="[166]"]
[2023/12/05 12:07:11.851 +08:00] [WARN] [prewrite.go:382] ["1pc failed and fallbacks to normal commit procedure"] [startTS=446103350620717112]
[2023/12/05 12:07:11.897 +08:00] [FATAL] [prewrite.go:393] ["one pc happened multiple times"] [startTS=446103350633562161] [stack="github.com/tikv/client-go/v2/txnkv/transaction.actionPrewrite.handleSingleBatch\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231201024404-0ff16620f6c0/txnkv/transaction/prewrite.go:393\ngithub.com/tikv/client-go/v2/txnkv/transaction.(*twoPhaseCommitter).doActionOnBatches\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231201024404-0ff16620f6c0/txnkv/transaction/2pc.go:1064\ngithub.com/tikv/client-go/v2/txnkv/transaction.(*twoPhaseCommitter).doActionOnGroupMutations\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231201024404-0ff16620f6c0/txnkv/transaction/2pc.go:1045\ngithub.com/tikv/client-go/v2/txnkv/transaction.(*twoPhaseCommitter).doActionOnMutations\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231201024404-0ff16620f6c0/txnkv/transaction/2pc.go:808\ngithub.com/tikv/client-go/v2/txnkv/transaction.(*twoPhaseCommitter).prewriteMutations\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231201024404-0ff16620f6c0/txnkv/transaction/prewrite.go:504\ngithub.com/tikv/client-go/v2/txnkv/transaction.(*twoPhaseCommitter).execute\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231201024404-0ff16620f6c0/txnkv/transaction/2pc.go:1513\ngithub.com/tikv/client-go/v2/txnkv/transaction.(*KVTxn).Commit\n\t/go/pkg/mod/github.com/tikv/client-go/v2@v2.0.8-0.20231201024404-0ff16620f6c0/txnkv/transaction/txn.go:508\ngithub.com/pingcap/tidb/pkg/store/driver/txn.(*tikvTxn).Commit\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/store/driver/txn/txn_driver.go:99\ngithub.com/pingcap/tidb/pkg/session.(*LazyTxn).Commit\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/txn.go:428\ngithub.com/pingcap/tidb/pkg/session.(*session).commitTxnWithTemporaryData\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:740\ngithub.com/pingcap/tidb/pkg/session.(*session).doCommit\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:621\ngithub.com/pingcap/tidb/pkg/session.(*session).doCommitWithRetry\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:874\ngithub.com/pingcap/tidb/pkg/session.(*session).CommitTxn\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:1001\ngithub.com/pingcap/tidb/pkg/session.autoCommitAfterStmt\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/tidb.go:298\ngithub.com/pingcap/tidb/pkg/session.finishStmt\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/tidb.go:260\ngithub.com/pingcap/tidb/pkg/session.(*execStmtResult).Close\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:2453\ngithub.com/pingcap/tidb/pkg/session.(*session).ExecRestrictedSQL.func1.1\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:2049\ngithub.com/pingcap/tidb/pkg/session.(*session).ExecRestrictedSQL.func1\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:2063\ngithub.com/pingcap/tidb/pkg/session.(*session).withRestrictedSQLExecutor\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:2027\ngithub.com/pingcap/tidb/pkg/session.(*session).ExecRestrictedSQL\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/session/session.go:2031\ngithub.com/pingcap/tidb/pkg/bindinfo.execRows\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/bindinfo/util.go:44\ngithub.com/pingcap/tidb/pkg/bindinfo.(*BindHandle).Update\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/bindinfo/handle.go:137\ngithub.com/pingcap/tidb/pkg/domain.(*Domain).LoadBindInfoLoop.(*Domain).globalBindHandleWorkerLoop.func1\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/domain/domain.go:1878\ngithub.com/pingcap/tidb/pkg/util.(*WaitGroupEnhancedWrapper).Run.func1\n\t/home/jenkins/agent/workspace/pingcap/tidb/periodics_integration_test/tidb/pkg/util/wait_group_wrapper.go:99"]
```
link 
https://do.pingcap.net/jenkins/blue/organizations/jenkins/pingcap%2Ftidb%2Fghpr_check2/detail/ghpr_check2/34865/pipeline/284

### 4. What is your TiDB version? (Required)

master branch

**Comment by qw4990** [comment]
Might be caused by https://github.com/pingcap/tidb/pull/49164

**Comment by Defined2014** [comment]
Seems fixed by #49164, so close this isssue.
